### PR TITLE
Adds a sort of EULA-style "I accept" entry gate for the server. 

### PR DESCRIPTION
**Pull request by EmperorJon**
Removes the MOTD from the chat window and pops it up when you join the server in a brand new window instead.
This window must be closed to continue to the normal menu, and basically consists of the MOTD + Rules + Map info + some useful links. Various config changes etc. made as a result, examples should be provided. Doesn't include actual configs or MOTD I tested with. When/if this goes through I'll throw them Zuh's way or whatever.

A global list (ew I know) is used to check which ckeys have pressed the button so you only see it once per round. It's also available as a Server Information verb in the OOC tab.

The idea behind this is that it can be used to show current info, event info, map-specific info, or whatever, as well as preventing "I didn't read the rules" as an excuse because you literally have to click a button on a page showing you the rules to continue.

Some newline etc. changes need to be removed due to me changing things a lot as I went but I'm interested in opinions on the verb at client level (preferred) and a suitable alternative to implementing a new client/Topic() with a parent call at the end to ensure it still works as usual.

### CODE REVIEW

File: code/modules/mob/new_player/server_info_window.dm [hunk 1]
```diff
@@ -0,0 +1,52 @@
+/client/verb/check_server_info()
+	set name = "Server Information"
+	set category = "OOC"
+
+	var/output = join_motd
+	var/list/close = list("Gotcha!","AFFIRMATIVE","EXCELSIOR!","I accept","The Pact Is Sealed","Continue","Take me to the fun!","Let us begin")
+	output += file2text("config/rules.html")
+	output += "<p><b>Map info:</b></p>"
+	output += using_map.motd
+	output += {"<br>
```
Comment (Techhead0) [inline code/modules/mob/new_player/server_info_window.dm]: If you're going to use a multi-line string, you can embed the other stuff in it with brackets. eg. 
```
{"[join_motd]
[file2text("config/rules.html")]
<p><b>Map info:</b></p>
[using_map.motd]<br>
	. . .
```
Saves a few string operations.

File: config/example/config.txt [hunk 2]
```diff
@@ -202,13 +202,19 @@ GUEST_BAN
 #SERVERURL server.net:port
 
 ## forum address
-# FORUMURL http://example.com
+FORUMURL http://example.com
```
Comment (Techhead0) [inline config/example/config.txt]: Probably shouldn't uncomment the `http://example.com` lines in the example config.

**Comment by Techhead0** [comment]
A separate (non-review) note: When this is merged and the server updated, the actual configs will need updating as well.

File: config/example/config.txt (hunk 2 above)
Comment (EmperorJon) [inline config/example/config.txt]: Was for testing, good catch.

**Comment by EmperorJon** [comment]
Yeah I have examples for someone to take a look at and decide if they need changes. Regarding the rebuilding-the-window, what would be the cleanest way to cache it?

**Comment by Techhead0** [comment]
As is, it's already being saved (at one per client). If you just check for an existing one in the verb, then you can skip making a new one.

If you want to globally cache one copy, then I'll have to get back to you later.

**Comment by EmperorJon** [comment]
Right, yeah. Overcomplicating things. Thanks.

File: code/modules/mob/new_player/server_info_window.dm [hunk 3]
```diff
@@ -0,0 +1,56 @@
+/client/verb/check_server_info()
+	set name = "Server Information"
+	set category = "OOC"
+
+	if (src.infowindow)  //If the user's already viewed the window before just load it again.
+		src.infowindow.open()
+		return
+
+	var/output = join_motd
```
Comment (Techhead0) [inline code/modules/mob/new_player/server_info_window.dm]: Put this line in the long string too.

**Comment by EmperorJon** [comment]
Voila.

**Comment by EmperorJon** [comment]
I have modified configs etc. that I can provide and/or give some insight into how _I_ personally did it, which is the images Zuh's seen. I can squash them if necessary.

**Comment by EmperorJon** [comment]
Squashing would have been a nightmare due to merges in the meantime so https://github.com/Yonaguni/EuropaStation/pull/563

**Pull request closed without merging**